src/Stack/Setup.hs

```diff
@@ -140,6 +140,8 @@ data SetupOpts = SetupOpts
     , soptsUpgradeCabal :: !Bool
     -- ^ Upgrade the global Cabal library in the database to the newest
     -- version. Only works reliably with a stack-managed installation.
+    , soptsInstallCabal :: !(Maybe Version)
+    -- ^ Install a specific version of Cabal.
```

Comment:
BTW, as an example `stack clean` uses the `Alternative` instance of the option parser:

```
Usage: stack clean ([PACKAGE] | [--full]) [--help]
```